### PR TITLE
ICU-21776: demo updates: re-enable Locale Explorer, fix source code formatting, etc

### PR DESCRIPTION
**Pull request by sven-oly**
This PR cleans up some compiler warnings in ICU4C demos, reformats white space, and updates the instructions for building the demos via Docker. The erroneous port number in the "docker run" command is corrected.

Most important, it adds a return to showTaggedArray. This prevents a call to ures_countArrayItems that causes the entire locale display to fail on many locales, e.g., [Aghem](http://localhost:8888/icu-bin/locexp?d_=en&_=agq_CM).

It also extends locales to allow 3 letter codes.

webdemo/collation/index.html is updated with new links.

More .h files are added to reduce warnings.

Note that there are more updates needed to remove compiler warnings and other fixes. In particular, I suggest that each demo clearly labels itself automatically with the CLDR & Unicode versions, ans done in the Unicode Browser demonstration program.

### CODE REVIEW

**Review by srl295 — DISMISSED** [review]
I'm giving a standing ovation here. Thanks for working on this.

**Comment by sven-oly** [comment]
PTAL - resolved conflicting lines in the Collation demo's links.

File: locexp/lxshow.cpp [hunk 1]
```diff
@@ -15,1154 +15,1154 @@ using namespace icu; // TODO: fix namespace
...
-    UBool userRequested = false; /* Did the user request this string? */
-    int32_t len = 0, len2, i;
-    UCollator *coll = NULL; /* build an actual collator */
-    UResourceBundle *array = NULL, *item = NULL;
 
-    array = ures_getByKey(rb, key, array, &status);
-    subStatus = status;
-  
-    s = ures_getStringByKey(array, "Sequence", &len, &subStatus);
+  UErrorCode status = U_ZERO_ERROR;
```
Comment (markusicu) [inline locexp/lxshow.cpp:35]: Why is this file changing from the normal indent of 4 to an indent of 2?

File: webdemo/collation/index.html [hunk 2]
```diff
@@ -96,7 +98,17 @@ <h1><a href="https://icu.unicode.org/">ICU</a> Collation Demo</h1>
 <option value="ja">ja (type=standard):   Japanese (Standard Sort Order)</option>
 <option value="ja@collation=unihan">ja-u-co-unihan:   Japanese (Radical-Stroke Sort Order)</option>
 <option value="ka">ka (type=standard):   Georgian (Standard Sort Order)</option>
-<option value="kk">kk (type=standard):   Kazakh (Standard Sort Order)</option>
+<option value="kk">kk (type=standard):   Kazakh (Standard Sort Order)</option>လါဒံၣ်စ့ဘၢၣ် ၈ သီ, ၂၀၂၁ နံၣ်, ခ့ၣ်အဲးစံၣ်
```
Comment (markusicu) [inline webdemo/collation/index.html:101]: Where does the random Myanmar text come from on several lines here? I don't think that the tool that writes this section could generate this text.

Please fix.

**Review by markusicu — DISMISSED** [review]
Mostly looks good, but one file apparently got mis-indented, and the collation demo locales list got an injection of text that doesn't belong.

File: locexp/lxshow.cpp (hunk 1 above)
Comment (sven-oly) [reply locexp/lxshow.cpp]: I've updated again to use indent of 4 spaces. Also removed lots of unneeded white space.

File: locexp/lxshow.cpp [hunk 3]
```diff
@@ -1576,61 +1576,70 @@ void show2dArrayWithDescription( LXContext *lx, UResourceBundle *rb, const char
...
+    // Todo - fix countArrayItems, which fails for many locales
+    return;
+
+    if (rb) {
+        rows = ures_countArrayItems(rb, key, &status);
+    } else {
+        fprintf(stderr, "showTaggedArray(1591). rb is NULL for locale %s\n", locale);
+        fflush(stderr);
+        return;
+    }
```
Comment (markusicu) [inline locexp/lxshow.cpp]: I see a new return that prevents execution of a fair bit of following code, and I also see a change in some of that following code. Is that change insufficient, and therefore we need the early return?

File: webdemo/collation/index.html (hunk 2 above)
Comment (sven-oly) [reply webdemo/collation/index.html]: Oops, bad edit.

Comment (srl295) [reply webdemo/collation/index.html]: @sven-oly  happens to me all the time …

File: locexp/lxshow.cpp (hunk 3 above)
Comment (sven-oly) [reply locexp/lxshow.cpp]: The new return avoids the crash in LocaleExplorer. This needs more work to fix the call to to ures_countArrayItems, which is where it died on many locales. I'll add a ticket to fix this so we don't forget.

Comment (sven-oly) [reply locexp/lxshow.cpp]: Added https://unicode-org.atlassian.net/browse/ICU-21581 indicating the need to fix this.

File: locexp/lxshow.cpp [hunk 4]
```diff
@@ -1576,61 +1576,70 @@ void show2dArrayWithDescription( LXContext *lx, UResourceBundle *rb, const char
...
+    const UChar *s  = 0;
+    int32_t v;
+    int32_t rows;
+    UBool bigString = false; /* is it big? */
+    UBool userRequested = false; /* Did the user request this string? */
+    int32_t len;
+    UResourceBundle *item = NULL;
+
+    // Todo - fix countArrayItems, which fails for many locales
+    return;
```
Comment (markusicu) [inline locexp/lxshow.cpp]:
```suggestion
    // Todo - fix countArrayItems, which fails for many locales
    // See https://unicode-org.atlassian.net/browse/ICU-21581
    return;
```

Comment (sven-oly) [reply locexp/lxshow.cpp]: Done

**Review by markusicu — COMMENTED** [review]
lgtm pse squash

**Review by markusicu — APPROVED** [review]
I see that this repo allows **squash-and-merge**. Feel free to use that instead of squashing separately, but please make sure to clean up the commit message.

**Comment by markusicu** [comment]
Oh, but "This branch cannot be rebased due to conflicts" :-/

**Pull request merged**